spec.html

```diff
@@ -30111,16 +30111,23 @@ <h1>Function.prototype.toString ( )</h1>
         <p>When the `toString` method is called, the following steps are taken:</p>
         <emu-alg>
           1. Let _func_ be the *this* value.
-          1. If Type(_func_) is Object and _func_ has a [[SourceText]] internal slot and _func_.[[SourceText]] is a sequence of Unicode code points and HostHasSourceTextAvailable(_func_) is *true*, then
+          1. If IsCallable(_func_) is *false*, throw a *TypeError* exception.
+          1. Assert: Type(_func_) is Object.
+          1. If _func_ has a [[SourceText]] internal slot and _func_.[[SourceText]] is a sequence of Unicode code points and HostHasSourceTextAvailable(_func_) is *true*, then
             1. Return CodePointsToString(_func_.[[SourceText]]).
-          1. If _func_ is a <emu-xref href="#sec-built-in-function-objects">built-in function object</emu-xref>, return an implementation-defined String source code representation of _func_. The representation must have the syntax of a |NativeFunction|. Additionally, if _func_ has an [[InitialName]] internal slot and _func_.[[InitialName]] is a String, the portion of the returned String that would be matched by |NativeFunctionAccessor?| |PropertyName| must be the value of _func_.[[InitialName]].
-          1. If Type(_func_) is Object and IsCallable(_func_) is *true*, return an implementation-defined String source code representation of _func_. The representation must have the syntax of a |NativeFunction|.
-          1. Throw a *TypeError* exception.
+          1. Let _sourceText_ be an implementation-defined sequence of Unicode code points that represents _func_ using the syntax of |NativeFunction|. If _func_ is a <emu-xref href="#sec-built-in-function-objects">built-in function object</emu-xref>, the portion of _sourceText_ corresponding to |NativeFunctionName| must be StringToCodePoints(_func_.[[InitialName]]).
+          1. Return CodePointsToString(_sourceText_).
         </emu-alg>
-
+        <emu-note>
+          <p>Every <emu-xref href="#sec-built-in-function-objects">built-in function object</emu-xref> has an [[InitialName]] internal slot with String value equal to the initial value of its *"name"* property, which for some functions is explicitly specified with contents that match the syntax of |ComputedPropertyName| rather than |LiteralPropertyName|. See <emu-xref href="#sec-ecmascript-standard-built-in-objects"></emu-xref>.</p>
+        </emu-note>
         <emu-grammar type="definition">
           NativeFunction :
-            `function` NativeFunctionAccessor? PropertyName[~Yield, ~Await]? `(` FormalParameters[~Yield, ~Await] `)` `{` `[` `native` `code` `]` `}`
+            `function` NativeFunctionName `(` FormalParameters[~Yield, ~Await] `)` `{` `[` `native` `code` `]` `}`
+
+          NativeFunctionName :
+            [empty]
```

Comment:
```suggestion
            `function` NativeFunctionName? `(` FormalParameters[~Yield, ~Await] `)` `{` `[` `native` `code` `]` `}`

          NativeFunctionName :
```

Reply:
I don't believe that would be compatible with "the portion of _sourceText_ corresponding to |NativeFunctionName| must be StringToCodePoints(_func_.[[InitialName]])" (because "the portion of _sourceText_ corresponding to |NativeFunctionName|" is an incoherent concept when |NativeFunctionName| is absent). This is one of the problems with the current text that I am attempting to fix.